lib/BC/Optimizer.cpp

```diff
@@ -46,81 +53,39 @@ namespace remill {
 void OptimizeModule(const remill::Arch *arch, llvm::Module *module,
                     std::function<llvm::Function *(void)> generator,
                     OptimizationGuide guide) {
-
-  llvm::legacy::FunctionPassManager func_manager(module);
-  llvm::legacy::PassManager module_manager;
-
-  auto TLI =
-      new llvm::TargetLibraryInfoImpl(llvm::Triple(module->getTargetTriple()));
-
-  TLI->disableAllFunctions();  // `-fno-builtin`.
-
-  llvm::PassManagerBuilder builder;
-  // TODO(alex): Some of the optimization passes that the builder adds still rely on typed pointers
-  // so we cannot use them. We should switch to using the new pass manager and choose which passes
-  // we want.
-  builder.OptLevel = 0;
-  builder.SizeLevel = 0;
-  builder.Inliner = llvm::createFunctionInliningPass(250);
-  builder.LibraryInfo = TLI;  // Deleted by `llvm::~PassManagerBuilder`.
-  builder.DisableUnrollLoops = false;  // Unroll loops!
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(16, 0)
-  builder.RerollLoops = false;
-#endif
-  builder.SLPVectorize = guide.slp_vectorize;
-  builder.LoopVectorize = guide.loop_vectorize;
-  builder.VerifyInput = guide.verify_input;
-  builder.VerifyOutput = guide.verify_output;
-  builder.MergeFunctions = false;
-
-  builder.populateFunctionPassManager(func_manager);
-  builder.populateModulePassManager(module_manager);
-  func_manager.doInitialization();
-  llvm::Function *func = nullptr;
-  while (nullptr != (func = generator())) {
-    func_manager.run(*func);
-  }
-  func_manager.doFinalization();
-  module_manager.run(*module);
+  OptimizeBareModule(module, guide);
```

Comment:
Not sure what we want to do about this i dont see an easy way to do single function inlining anymore i guess we could probably manually call llvm::InlineFunction etc but that seems excessive for something only called in the trace lifter idk